### DISCUSSION TITLE
Access env variables from next.config.js outside of Next.js app?

### DISCUSSION THREAD

**Discussion by nemanjam** [vercel/next.js#33017] [Help]
I have Next.js app with custom server and static folder for uploads. I defined environment variable for path for uploads in `next.config.js` in `env` node and I need that variable inside Next.js app code but also in custom server and seed script and those three are separate build contexts. How can I access it in custom server and seed script?

I could define it in .env file and then reference that in `next.config.js` but I would prefer to have less data in .env file and define it in `next.config.js`.

I know for `loadEnvConfig(process.cwd())` but that just loads vars from `.env.*` files, not from `next.config.js`.

```
// next.config.js
module.exports = {
  ...
  env: {
    NEXT_PUBLIC_UPLOADS_PATH: '/uploads/',
  },
}
```

**Comment by dmurawsky** [comment]
Bump
